server/src/test/java/io/crate/integrationtests/IndexUpgraderTest.java

```diff
@@ -117,4 +119,420 @@ public void test_index_upgraders_fix_column_positions() throws Exception {
+        //mappings = {ImmutableOpenMap@12031} "[default=>org.elasticsearch.cluster.metadata.MappingMetadata@bb15917e]"
+        // map = {ObjectObjectHashMap@12036} "[default=>org.elasticsearch.cluster.metadata.MappingMetadata@bb15917e]"
+        //  keys = {Object[9]@12038}
+        //  values = {Object[9]@12039}
+        //   0 = {MappingMetadata@12040}
+        //    source = {CompressedXContent@12041} "{"default":{"dynamic":"strict","_meta":{"partitioned_by":[["s","keyword"]]},"properties":{"s":{"type":"keyword","index":false,"position":1,"length_limit":1,"blank_padding":true}}}}"
+        // NOTE: m1.t1/Xgrwv6giTTKgdh57L6g6zA could not be queried due to an NPE thrown.
+
+        execute("select * from m1.s1 order by t");
+        assertThat(printedTable(response.rows())).isEqualTo("false\nfalse\ntrue\n");
+        execute("show create table m1.s1");
+        assertThat(printedTable(response.rows())).contains(
+            List.of(
+                """
+                    CREATE TABLE IF NOT EXISTS "m1"."s1" (
+                       "t" BOOLEAN
+                    )
+                    """,
+                "refresh_interval = 300",
+                "PARTITIONED BY (\"t\")")
+        );
+
+        execute("select * from m1.broken_s1");
+        assertThat(printedTable(response.rows())).isEmpty();
+        execute("show create table m1.broken_s1");
+        assertThat(printedTable(response.rows())).contains(
+            List.of(
+                """
+                    CREATE TABLE IF NOT EXISTS "m1"."broken_s1" (
+                       "s" CHARACTER(1)
+                    )
+                    """,
+                "refresh_interval = 400",
+                "PARTITIONED BY (\"s\")"
+            )
+        );
+
+        execute("select * from m1.broken_t1");
+        assertThat(printedTable(response.rows())).isEmpty();
+        execute("show create table m1.broken_t1");
+        assertThat(printedTable(response.rows())).contains(
+            List.of(
+                """
+                    CREATE TABLE IF NOT EXISTS "m1"."broken_t1" (
+                       "s" CHARACTER(1)
+                    )
+                    """,
+                "PARTITIONED BY (\"s\")"
+            )
+        ).doesNotContain("refresh_interval = 400");
```

Comment:
There are two recovered tables (without data) one was from index metadata and the other was from template metadata.